src/Nethermind/Nethermind.Core/Extensions/MemoryExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Nethermind.Core.Extensions;
+
+public static class MemoryExtensions
+{
+    /// <summary>
+    /// Try to get the underlying array if possible to prevent copying.
+    /// </summary>
+    /// <param name="memory"></param>
+    /// <returns></returns>
+    public static byte[]? FasterToArray(this in Memory<byte>? memory)
+    {
+        if (memory == null) return null;
+
+        return memory.Value.FasterToArray();
+    }
+
+    public static byte[] FasterToArray(this in Memory<byte> memory)
```

Comment:
Naming could cause confusion, normally you expect a copy. 
Minor: is the code layout readable here? Would prefer `?` operator.

Reply:
What do you have in mind? `AsArray`, `AsArrayOrCopy`?

Reply:
I like `.AsArray()`